x/dsmr/node.go

```diff
@@ -133,28 +141,49 @@ func (n *Node[T]) BuildChunk(
 		return Chunk[T]{}, err
 	}
 
-	request := &dsmr.GetChunkSignatureRequest{
-		Chunk: packer.Bytes,
+	unsignedMsg, err := warp.NewUnsignedMessage(n.networkID, n.chainID, packer.Bytes)
+	if err != nil {
+		return Chunk[T]{}, fmt.Errorf("failed to initialize unsigned warp message: %w", err)
+	}
+	msg, err := warp.NewMessage(
+		unsignedMsg,
+		&warp.BitSetSignature{
+			Signature: [bls.SignatureLen]byte{},
+		},
+	)
+	if err != nil {
+		return Chunk[T]{}, fmt.Errorf("failed to initialize warp message: %w", err)
 	}
 
-	for _, validator := range n.validators {
-		done := make(chan struct{})
-		onResponse := func(context.Context, ids.NodeID, *dsmr.GetChunkSignatureResponse, error) {
-			defer close(done)
-			// TODO generate chunk cert
-		}
+	validators := make([]acp118.Validator, 0, len(n.validators))
+	for _, v := range n.validators {
+		validators = append(validators, acp118.Validator{
+			PublicKey: v.PublicKey,
+			Weight:    v.Weight,
+			NodeID:    v.NodeID,
+		})
+	}
 
-		if err := n.getChunkSignatureClient.AppRequest(ctx, validator.NodeID, request, onResponse); err != nil {
-			return Chunk[T]{}, err
-		}
+	aggregatedMsg, aggregatedStake, totalStake, err := n.chunkSignatureAggregator.AggregateSignatures(
+		ctx,
+		msg,
+		nil,
+		validators,
+		n.quorumNum,
+		n.quorumDen,
+	)
+	if err != nil {
+		return Chunk[T]{}, fmt.Errorf("failed to aggregate signatures: %w", err)
+	}
 
-		<-done
+	if warp.VerifyWeight(aggregatedStake, totalStake, n.quorumNum, n.quorumDen) != nil {
+		return Chunk[T]{}, fmt.Errorf("failed to replicate to sufficient stake: %w", err)
 	}
 
-	chunkCert := ChunkCertificate{
+	chunkCert := ChunkCertificate[T]{
 		ChunkID:   chunk.id,
 		Expiry:    chunk.Expiry,
-		Signature: NoVerifyChunkSignature{},
+		Signature: aggregatedMsg.Signature.(*warp.BitSetSignature),
```

Comment:
Can we move this typecheck above and return an error if it fails to future proof?